File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.py
def check(s):
    stack = []

    ans = False
    for i in s:
        if i == '[' or i == '(' or i == '{':
            stack.append(i)
            ans = True

        elif len(stack) and i == ']' and stack[-1] == '[':
            stack.pop()

        elif len(stack) and i == ')' and stack[-1] == '(':
            stack.pop()

        elif len(stack) and i == '}' and stack[-1] == '{':
            stack.pop()

    if len(stack) or not ans:
        return False
    elif len(stack) == 0 or ans:
        return True
    
    
def solution(s):
    
    r = ''
    ans = 0
    
    for i in range(len(s)):
        r = s[i:] + s[:i]
        
        if check(r):
            ans += 1
            
    return ans

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public static boolean check(String s) {
        Stack<Character> stack = new Stack<>();
        
        for (char ch : s.toCharArray()) {
            if (ch == '(' || ch == '{' || ch == '['){
                stack.add(ch);
            } else if (ch == ')'){
                if (stack.empty() || stack.peek() != '(') {
                    return false;
                }
                stack.pop();
            } else if (ch == '}'){
                if (stack.empty() || stack.peek() != '{') {
                    return false;
                }
                stack.pop();
            } else if (ch == ']'){
                if (stack.empty() || stack.peek() != '[') {
                    return false;
                }
                stack.pop();
            } 
        }
        System.out.println(stack);
        return stack.empty() ? true : false;
    }
    
    public static String shiftLeft(String s) {
        return s.substring(1) + s.charAt(0);
    }
    public static int solution(String s) {
        int answer = 0;
        for (int i = 0; i < s.length(); i++){
            s = shiftLeft(s);
            if (check(s)) answer++;
        } 
        
        return answer;
    }
}